app/metrics.py

```diff
@@ -80,6 +80,10 @@ def get_full_current_metrics(w3, beacon, beacon_spec, partial_metrics) -> PoolMe
     logging.info(
         f'Lido validators\' sum. balance on Beacon: {full_metrics.beaconBalance} wei or {full_metrics.beaconBalance/1e18} ETH'
     )
+
+    block_number = beacon.get_block_by_beacon_slot(slot)
+    partial_metrics.withdrawalVaultBalance = w3.eth.get_balance(withdrawal_vault_address, block_identifier=block_number)
```

Comment:
it's better to use blockhash. I guess the issue here, is that we need to include a workaround for web3 for this, right?

Reply:
I would prefer to make minimal changes to the current oracle, and this trade-off (don't patch web3) looks reasonable